tests/posterior_nn_test.py

```diff
@@ -137,13 +138,67 @@ def test_batched_mcmc_sample_log_prob_with_different_x(
     x_o = ones(num_dim) if x_o_batch_dim == 0 else ones(x_o_batch_dim, num_dim)
 
     posterior = inference.build_posterior(
-        mcmc_method="slice_np_vectorized", mcmc_parameters=mcmc_params_fast
+        sample_with="mcmc",
+        mcmc_method="slice_np_vectorized",
+        mcmc_parameters=mcmc_params_fast,
     )
 
-    samples = posterior.sample_batched((10,), x_o)
+    samples = posterior.sample_batched(
+        (10,),
+        x_o,
+        init_strategy=init_strategy,
+        num_chains=2,
+    )
 
     assert (
         samples.shape == (10, x_o_batch_dim, num_dim)
         if x_o_batch_dim > 0
         else (10, num_dim)
     ), "Sample shape wrong"
+
+
+@pytest.mark.mcmc
+@pytest.mark.parametrize(
+    "snlre_method",
+    [SNLE_A, SNRE_A, SNRE_B, SNRE_C, SNPE_C],
+)
+def test_batched_mcmc_sample_log_prob_with_different_x(
```

Comment:
On a first look, it seems this test is a superset of the above test, no? Maybe they can be combined?

Reply:
agreed, will combine.